src/v/pandaproxy/schema_registry/json.cc

```diff
@@ -499,10 +520,13 @@ json_type_list normalized_type(json::Value const& v) {
     return ret;
 }
 
-// helper to convert a boolean to a schema
+// helper to convert a boolean to a schema, and traverse references
 json::Value::ConstObject
-get_schema(schema_context const&, json::Value const& v) {
+get_schema(schema_context const& ctx, json::Value const& v) {
     if (v.IsObject()) {
+        if (auto it = v.FindMember("$ref"); it != v.MemberEnd()) {
+            return ctx.resolve(as_string_view(it->value)).GetObject();
```

Comment:
this should be the ok for draft4/7 ($ref overwrites the schema), but reading the spec seems like for 2019/20 the result should be merged on top of the schema. 

```
A schema can reference another schema using the $ref keyword. The value of $ref is a URI-reference that is resolved against the schema's [Base URI](https://json-schema.org/understanding-json-schema/structuring#base-uri). When evaluating a $ref, an implementation uses the resolved identifier to retrieve the referenced schema and applies that schema to the [instance](https://json-schema.org/learn/glossary#instance).
Draft-specific info
In Draft 4-7, $ref behaves a little differently. When an object contains a $ref property, the object is considered a reference, not a schema. Therefore, any other properties you put in that object will not be treated as JSON Schema keywords and will be ignored by the validator. $ref can only be used where a schema is expected.
```

Then again, I suspect our reference implementation does not implement the merge behavior.
I'll check